### PR TITLE
Test for SNAP-2235

### PR DESCRIPTION
**Pull request by kneeraj**
## Changes proposed in this pull request

Added backup and restore comand line tools test. The newly added file should be used to add any command line testing we do.

## Patch testing

This is itself a test.

## ReleaseNotes.txt changes

No.

## Other PRs 

https://github.com/SnappyDataInc/snappy-store/pull/373

### CODE REVIEW

File: core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+import java.io._
+import java.sql.{Connection, DriverManager}
+
+import org.apache.commons.io.output.TeeOutputStream
+import org.apache.spark.sql.collection.Utils
+
+import scala.sys.process._
+
+/**
+  * Created by kneeraj on 12/3/18.
```
Comment (rishitesh) [inline core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala]: Remove the author info.

File: core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+  def getJdbcConnection(netPort: Int): Connection = {
+    val driver = "io.snappydata.jdbc.ClientDriver"
+    Utils.classForName(driver).newInstance
+    var url: String = "jdbc:snappydata://localhost:" + netPort + "/"
+    DriverManager.getConnection(url)
+  }
+
+  // CWD will be assumed the same for all command which is $snappyHome
+  def executeCommand(command: String): (String, String) = {
```
Comment (rishitesh) [inline core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala]: executeCommand() method can be moved to SnappyTestRunner for code reuse. Same is the case with getJdbcConnection()

File: core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+      while (rs22.next()) {
+        cnt = cnt + 1
+        val v1 = rs22.getInt(1)
+        val v2 = rs22.getInt(2)
+        assert(v1 === 1 || v1 === 2 || v1 === 5 || v1 === 6 || v1 === 100 || v1 === 200)
+      }
+      assert(cnt === 6)
+    } finally {
+      debugWriter.close()
+      executeCommand(s"rm -rf $snappyHome/backup*")
```
Comment (rishitesh) [inline core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala]: Can we create all backup artifacts under another folder preferable similar to test name? If for some reason the test gets killed we would know it's not part of SnappyHome. Not a high priority, but IMO preferable.

Comment (kneeraj) [reply core/src/test/scala/io/snappydata/CommandLineToolsSuite.scala]: Moved to /tmp/

**Pull request merged**